api/preprints/serializers.py

```diff
@@ -75,184 +83,105 @@ class PreprintSerializer(JSONAPISerializer):
         }
     )
 
-    contributors = RelationshipField(
-        related_view='nodes:node-contributors',
-        related_view_kwargs={'node_id': '<pk>'},
-        related_meta={'count': 'get_contrib_count'},
-    )
-
     class Meta:
         type_ = 'preprints'
 
     def get_preprint_url(self, obj):
-        kwargs = {
-            'node_id': obj._id,
-            'version': self.context['request'].parser_context['kwargs']['version']
-        }
-        return absolute_reverse('preprints:preprint-detail', kwargs=kwargs)
+        return absolute_reverse('preprints:preprint-detail', kwargs={'preprint_id': obj._id, 'version': self.context['request'].parser_context['kwargs']['version']})
 
     def get_absolute_url(self, obj):
         return self.get_preprint_url(obj)
 
     def get_doi_url(self, obj):
-        return 'https://dx.doi.org/{}'.format(obj.preprint_doi) if obj.preprint_doi else None
+        return 'https://dx.doi.org/{}'.format(obj.preprint_article_doi) if obj.preprint_article_doi else None
 
-    def create(self, validated_data):
-        node = Node.load(validated_data.pop('_id', None))
-        if not node:
-            raise exceptions.NotFound('Unable to find Node with specified id.')
+    def update(self, preprint, validated_data):
+        from website.models import Node
+        assert isinstance(preprint, PreprintService), 'You must specify a valid preprint to be updated'
```

Comment:
I don't _think_ assert is the proper way to do this here. Don't they get stripped out in production code? I believe if statements are preferred.  http://www.django-rest-framework.org/api-guide/validators/#writing-custom-validators

Or is this just a thing that is meant to catch developers and not API consumers.